java/client/src/main/java/glide/api/BaseClient.java

```diff
@@ -703,6 +690,78 @@ protected Map<String, Object> handleLcsIdxResponse(Map<String, Object> response)
         return response;
     }
 
+    /**
+     * @param response A Protobuf response
+     * @return A map of a map of <code>String[][]</code>
+     */
+    protected Map<String, Map<String, String[][]>> handleXReadResponse(Response response)
+            throws RedisException {
+        Map<String, Object> mapResponse = handleMapOrNullResponse(response);
+        if (mapResponse == null) {
+            return null;
+        }
+        return mapResponse.entrySet().stream()
+                .collect(
+                        Collectors.toMap(
+                                Map.Entry::getKey,
+                                e -> castMapOf2DArray((Map<String, Object[][]>) e.getValue(), String.class)));
+    }
+
+    /** Converts array types in the response from `XINFO STREAM` to be more user-friendly. */
+    @SuppressWarnings("unchecked")
+    protected Map<String, Object> handleXInfoStreamResponse(Map<String, Object> map) {
+        // convert entries from `Object[][]` to `String[][]` inside a map
+        for (var keyName : List.of("entries", "first-entry", "last-entry")) {
+            if (!map.containsKey(keyName)) continue;
+            var entries = (Map<String, Object[][]>) map.get(keyName);
+            map.put(keyName, castMapOf2DArray(entries, String.class));
+        }
+        for (var keyName : List.of("groups", "consumers")) {
+            if (!map.containsKey(keyName)) continue;
+            var value = map.get(keyName);
+            // simple response (no `FULL` keyword) - groups mapped to a number, not to a collection
+            if (value instanceof Long) {
+                return map;
+            }
+            // convert `Object[]` to `Map[]`
+            value = castArray((Object[]) value, Map.class);
+            for (var subMap : (Map<String, Object>[]) value) {
+                // recursively inline update the map
+                handleXInfoStreamResponse(subMap);
+            }
+            map.put(keyName, value);
+        }
+        return map;
+    }
+
+    /** Converts array types in the response from `XINFO STREAM` to be more user-friendly. */
+    @SuppressWarnings("unchecked")
+    protected Map<GlideString, Object> handleXInfoStreamResponseGlideString(
+            Map<GlideString, Object> map) {
+        // convert entries from `Object[][]` to `GlideString[][]` inside a map
+        for (var keyName : List.of(gs("entries"), gs("first-entry"), gs("last-entry"))) {
+            if (!map.containsKey(keyName)) continue;
+            var entries = (Map<GlideString, Object[][]>) map.get(keyName);
+            map.put(keyName, castMapOf2DArrayGlideString(entries, GlideString.class));
+        }
+        for (var keyName : List.of(gs("groups"), gs("consumers"))) {
```

Comment:
We always convert simplestrings to strings.  These shouldn't be GlideStrings. Reference:
"groups": https://github.com/aws/glide-for-redis/blob/main/glide-core/src/client/value_conversion.rs#L784
"consumers": https://github.com/aws/glide-for-redis/blob/main/glide-core/src/client/value_conversion.rs#L833

Reply:
so either this doesn't do anything, or the value_conversion isn't doing it's think